### ISSUE TITLE
FEST test for restore Trezor wallet

### ISSUE THREAD

**Issue by gary-rowe** [bitcoin-solutions/multibit-hd#328]
Require a FEST test to cover the "Restore Trezor at warm start" use case defined in #303.

This issue also covers the work required to implement the use case.

**Comment by bitcoin-solutions** [comment]
Pay attention to the PREV button in the restore wizard as it can lead to a 'enter wallet words' screen which is not appropriate for a Trezor hard wallet

**Comment by gary-rowe** [comment]
After discussion note that we should be able to remove the "Trezor select options" screen. This is up to Jim to do as part of completing this issue.

**Comment by gary-rowe** [comment]
Don't start this until #325 is closed. There are a lot of supporting framework changes occurring as part of the hardening process.

**Comment by bitcoin-solutions** [comment]
Also remember that loading from rolling backups is currently not working 100% as BackupManager is being initialised in MainController too late.

**Comment by bitcoin-solutions** [comment]
The rolling backups now load correctly with the user getting feedback on the wallet load report:
![screen shot 2015-01-15 at 15 11 34](https://cloud.githubusercontent.com/assets/5002074/5759899/f5860e3e-9cc8-11e4-84a7-5d91952c7661.png)

**Comment by bitcoin-solutions** [comment]
@gary-rowe I have got almost all of the "Trezor restore - there is a local backup " FEST working but for the mock client is not returning the entropy.
It is MultiBitFestTest#verifyRestoreTrezorWithLocalBackup

It works in real life, but the mock client needs to return WalletFixture.ABANDON_SEED_PASSWORD (as bytes in the payload) when asked. This is after the unlock confirm. At the moment the wallet does not load after a restore in the FEST test as it does not have this. The entropy is used as the password to decrypt the backup.

Could you give me a hand on this as I'm not familiar with the mock client ? 
Thanks

**Comment by gary-rowe** [comment]
The mock fixtures can be a bit fiddly and are contained in `TrezorHardwareWalletClientFixtures` for client and `HardwareWalletEventFixtures` for "user input" events (spontaneous events issued from the device). 

In this case look at `newCipherKeySuccess` to see the payload that will be given in response to a successful GetCipherKey request. That hex should represent the "abandon" wallet unlock key, but you could add a different method (and modify the subsequent uses throughout the two fixture factories) to deliver a different value for a different overall use case.

**Comment by bitcoin-solutions** [comment]
You'll have to give me a hand with this as it is taking far too long.
I have checked in the diagnostics I have so far on MultiBitFestTest#verifyRestoreTrezorWithLocalBackup
Somewhere the HardwareWalletContext is being lost and it is definitely a test artifact as it does not happen in real life.

After the initial PIN matrix and unlock the wallet loads, so the entropy is there. But I've logged in RestoreTrezorWarmStartRequirements, immediately after I fire the message with the entopy in, and it's not there. 

Then when the restore button is pressed and the backup is decrypted (with the password derived  from the entropy!) successfully. Then the wallet does not load because the HardwareWalletContext does not have the entropy.

**Comment by gary-rowe** [comment]
OK, I'll finish it off over the weekend.

**Comment by bitcoin-solutions** [comment]
Brilliant thanks.
I think it's one of those 'the state is here - no it's here' or maybe a resetting problem

**Comment by gary-rowe** [comment]
As usual this was a lot more work than it first appeared. It highlighted a significant problem in the hardware wallet FEST fixture API that made it too confusing to reason about. Consequently I've refactored it into something a lot simpler and more robust.

During the debugging and general fixing phase I uncovered a few subtle bugs to do with language switching and timing issues on startup. After sorting them out all FEST tests are working.

To verify this one, just run `MultiBitHDFestTest.verifyRestoreTrezorWithLocalBackup`

Ready for review and close.
